### PR TITLE
Implemented read

### LINKED ISSUE #1: Could implement constructor

### LINKED ISSUE #4: Must implement run

### LINKED ISSUE #9: Could extend superclass
So, when consuming API endpoints a reasonably standard block of code is used.  This should be used in an inheritance fashion.

### PR DESCRIPTION
**Pull request by sqweelygig**
* created a basic poll to the front api
closes #9
closes #4
closes #1

### CODE REVIEW

File: src/front.coffee [hunk 1]
```diff
@@ -1,28 +1,72 @@
...
-		@robot.logger.info 'Reply'
+	getOptions: (url) ->
+		{
+			url: url
+			headers: {
+				Accept: 'application/json'
+				Authorization: 'Bearer ' + process.env.HUBOT_FRONT_API_TOKEN
+			}
+			qs: {
+				before: moment().unix()
```
Comment (Page-) [inline src/front.coffee]: Whilst I do like momentjs, I think this is more standard as `before: Date.now()`

File: src/front.coffee [hunk 2]
```diff
@@ -1,28 +1,72 @@
...
+	extractResults: (obj) -> obj._results
 
-	send: (envelope, strings...) ->
-		@robot.logger.info 'Send'
+	poll: =>
+		@getUntil(
+			@getOptions process.env.HUBOT_FRONT_API_URL + '/events'
+			@processMessage
+			(obj) => obj.emitted_at > ( @robot.brain.get('AdapterFrontLastKnown') ? moment().subtract(parseInt(process.env.HUBOT_FRONT_SYNC_HISTORY), 'h').unix() )
+			(err, res) =>	if not err then @robot.brain.set('AdapterFrontLastKnown', res[0].emitted_at)
```
Comment (Page-) [inline src/front.coffee]: There's a tab after the `=>` rather than a space

File: src/front.coffee [hunk 3]
```diff
@@ -1,28 +1,72 @@
...
+
+	extractResults: (obj) -> obj._results
 
-	send: (envelope, strings...) ->
-		@robot.logger.info 'Send'
+	poll: =>
+		@getUntil(
+			@getOptions process.env.HUBOT_FRONT_API_URL + '/events'
+			@processMessage
+			(obj) => obj.emitted_at > ( @robot.brain.get('AdapterFrontLastKnown') ? moment().subtract(parseInt(process.env.HUBOT_FRONT_SYNC_HISTORY), 'h').unix() )
```
Comment (Page-) [inline src/front.coffee]: It would be nicer to do the `parseInt(process.env.HUBOT_FRONT_SYNC_HISTORY)` just once and store it in a module level constant

Comment (sqweelygig) [reply src/front.coffee]: This is only done once, but that isn't clear.  Am refactoring.

File: src/front.coffee [hunk 4]
```diff
@@ -1,28 +1,76 @@
...
+				@getOptions message.conversation._links.related.inboxes
+				@tellHubot message
+			)
 
+	tellHubot: (message) -> (inbox) =>
+		if message.target?.data?.author?
+			author = @robot.brain.userForId(
+				message.target.data.author.id
+				{ name: message.target.data.author.username, room: inbox.id }
+			)
```
Comment (CameronDiver) [inline src/front.coffee:56]: We normally drop brackets for multiline function calls in coffee

File: src/front.coffee [hunk 5]
```diff
@@ -1,28 +1,76 @@
...
+		{
+			url: url
+			headers: {
+				Accept: 'application/json'
+				Authorization: 'Bearer ' + process.env.HUBOT_FRONT_API_TOKEN
+			}
+			qs: {
+				before: Date.now()
+			}
+		}
```
Comment (CameronDiver) [inline src/front.coffee]: These braces aren't necessary, although I guess it comes down to personal preference.

**Review by CameronDiver — APPROVED** [review]
I added a couple of style comments, but other than that the code looks sane enough to me. Note that I haven't been able to test it, so I can't comment on the efficacy.

**Pull request merged**